smart_control/dataset/dataset.py

```diff
@@ -1,96 +1,469 @@
+        |  2 | rooms/9028472496    | buildings/3616672508 | SB1-2-2D4H         |      0 |           1 |       2 | ['2696593986887004']                     |           1 |
+        |  3 | rooms/9028558963    | buildings/3616672508 | SB1-2-2D4B         |      0 |           1 |       2 | ['2696593986887004']                     |           1 |
+        |  4 | rooms/9028483453    | buildings/3616672508 | SB1-2-2D4G         |      0 |           1 |       2 | ['2696593986887004']                     |           1 |
+    """
+    # pylint: enable=line-too-long
+    df = pd.DataFrame(self.zone_infos)
+    df["n_devices"] = df["devices"].apply(len)
+    return df
+
+
+class BuildingDatasetPartition:
+  """A helper class for handling a specific dataset partition.
+
+  Args:
+    building_dataset (BuildingDataset): The building dataset.
+    partition_id (str): The identifier of a partition in the specified dataset
+      (e.g. "2022_a").
+
+  Example:
+    ```python
+    ds = BuildingDataset(building_id='sb1', download=True)
+    partition = BuildingDatasetPartition(
+       building_dataset=ds, partition_id='2022_a'
+    )
+    ```
+  """
+
+  def __init__(self, building_dataset: BuildingDataset, partition_id: str):
+    self.ds = building_dataset
+    self.partition_id = partition_id
+
+    if self.partition_id not in self.ds.partition_ids:
+      raise ValueError(f"Invalid partition: {self.partition_id}.")
+
+  @property
+  def partition_dirpath(self):
+    return os.path.join(self.ds.tabular_dirpath, self.ds.building_id, self.partition_id)  # pylint:disable=line-too-long
+
+  @property
+  def data_filepath(self):
+    return os.path.join(self.partition_dirpath, "data.npy.npz")
+
+  @cached_property
+  def data(self) -> np.lib.npyio.NpzFile:
+    """Time-series data for the dataset partition.
+
+    This property returns an `np.lib.npyio.NpzFile` object, which allows
+      dictionary-like access to NumPy arrays stored within a compressed
+      `.npz` archive. The arrays contain time-series data, where the first
+      dimension typically represents the time steps.
+
+    Returns:
+      A dictionary-like numpy object with the following keys:
+
+        - `'observation_value_matrix'`
+        - `'action_value_matrix'`
+        - `'reward_value_matrix'`
+        - `'reward_info_value_matrix'`
+
+    """
+    return np.load(self.data_filepath)
+
+  @property
+  def metadata_filepath(self):
+    return os.path.join(self.partition_dirpath, "metadata.pickle")
+
+  @cached_property
+  def metadata(self) -> dict:
+    """Metadata describing the partition [data](./#smart_control.dataset.dataset.BuildingDatasetPartition.data).
+
+    Returns:
+      A dictionary containing the following keys:
+
+        - `'action_ids'`: dictionary mapping action string identifiers to
+          integer indices, for example: `{'action_id@setpoint': 0}`
```

Comment:
Todo: verify the meaning of the key names for `action_ids`, `observation_ids`, and `reward_ids`